### PR TITLE
[14.0][mig] base_ubl_payment

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by dsolanki-initos** [comment]
PR fails as the depended module PR (https://github.com/OCA/community-data-files/pull/100) still left to merge in 14.0

**Comment by rconjour** [comment]
@dsolanki-initos i see that the changes of PR #100 are already in the 14.0 branch. Can you please link with the right PR? I think if you extend `oca-dependencies.txt` with `community-data-files` the tests will pass.

**Comment by rconjour** [comment]
@dsolanki-initos seems like my previous comment was not right to solve the error. There is indeed still a depenency on this pr: https://github.com/OCA/community-data-files/pull/100 . I missed that that one wasn't merged yet.

To let your tests pass, you can extend `test-requirements.txt` with: `odoo14-addon-account_payment_unece @ git+https://github.com/akretion/community-data-files@14-mig-account_payment_unece#subdirectory=setup/account_payment_unece
`

Please keep that in a sepperate commit, that we can remove when the tests have passed!

**Comment by rconjour** [comment]
@dsolanki-initos we got https://github.com/OCA/community-data-files/pull/100 merged through the superseding PR: https://github.com/OCA/community-data-files/pull/107 , feel free to delete your last commit so that the tests can become green.

**Comment by dsolanki-initos** [comment]
Hi @rconjour I have updated the commits.

**Review by rconjour — APPROVED** [review]
LGTM - Thanks!

File: base_ubl_payment/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+{
+    "name": "Base UBL Payment",
+    "version": "14.0.1.0.0",
+    "category": "Hidden",
+    "license": "AGPL-3",
+    "summary": "Payment-related code for Universal Business Language (UBL)",
+    "author": "Akretion,Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/edi",
+    "depends": ["account_payment_unece", "base_ubl"],
+    "external_dependencies": {"python": ["lxml"]},
```
Comment (StefanRijnhart) [inline base_ubl_payment/__manifest__.py]: @astirpe is there any reason that you added this here: https://github.com/OCA/edi/pull/155/files#diff-304f9a575a9d52d595fac0aa80402d1e589ef778600d081546c20ff8468a0fc1R13? `lxml` is a standard Odoo requirement, so I do not expect it to be listed as an addon dependency. I would suggest to remove it if it is redundant indeed.

**Comment by StefanRijnhart** [comment]
@dsolanki-initos thanks for the update. You have to enable the pre-commit hook in your local git clone and recommit (amend) to have pre-commit remove the lxml dependency from `requirements.txt` in your commit.

**Review by rconjour — APPROVED** [review]
Seems like the feedback of @StefanRijnhart is implemented. Approved this PR again.

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

**Comment by StefanRijnhart** [comment]
This PR's Travis was happy to have `lxml` in the requirements.txt, but runbot's Travis fails on it
```
  # generated from manifests external_dependencies
-lxml
 PyPDF2
```
I see you did not take my suggestion to remove it (or maybe I wasn't clear). Can you try that?

**Comment by dsolanki-initos** [comment]
@StefanRijnhart can you please try to merge it again

**Comment by StefanRijnhart** [comment]
Thanks for the comment, we're waiting for this to be merged but I don't check in every day to see if the code was updated ;-)

BTW the lxml entry is still here: https://github.com/OCA/edi/pull/333/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R2. Is it perhaps readded by your local pre-commit hook? Perhaps there is a version difference that causes this.

Let's give it a try.

/ocabot merge nobump

**Comment by StefanRijnhart** [comment]
That looks like a bot/github gitch. Again!

/ocabot merge nobump

**Comment by StefanRijnhart** [comment]
@dsolanki-initos nope, it's still the problem that `lxml` is present in `requirements.txt` in your branch, and pre-commit does not want it there. Do you know how to solve it? Perhaps a reinstall (on a fresh git clone) of the pre-commit hook, or a rebase against edi/14.0 will solve the problem?

**Comment by dsolanki-initos** [comment]
> @dsolanki-initos nope, it's still the problem that `lxml` is present in `requirements.txt` in your branch, and pre-commit does not want it there. Do you know how to solve it? Perhaps a reinstall (on a fresh git clone) of the pre-commit hook, or a rebase against edi/14.0 will solve the problem?

Okay I'll try it by rebasing it with edi/14.0 .

**Comment by dsolanki-initos** [comment]
@StefanRijnhart I have rebased this branch with edi/14.0 .

**Comment by StefanRijnhart** [comment]
@dsolanki-initos thanks! `lxml` has now disappeared from requirements.txt. Only thing is, we now have a conflict in `test-requirements`, so I have to bother you for Yet Another Rebase :sob:

**Pull request merged**